File: payloads/go/write_to_bin.go
package main

import (
	"fmt"
	"os"
)

func main() {
	err := os.WriteFile("/bin/evil", []byte("#!/bin/sh\necho pwned"), 0755)
	if err != nil {
		panic(err)
	}
	fmt.Println("ESCAPED")
}

File: payloads/cpp/write_to_bin.cpp
#include <cstdio>
#include <fstream>

int main() {
    std::ofstream f("/bin/evil");
    if (!f.is_open()) return 1;
    f << "#!/bin/sh\necho pwned\n";
    f.close();
    std::printf("ESCAPED\n");
    return 0;
}
